### ISSUE TITLE
Add custom MSBuild loggers

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/KoreBuild#183]
Ideas for important logger features:

 - Log TeamCity blocks and flow ID's. Minimizing output (we currently spit out ~100 MB of text).
 - Log Travis CI folds (aka blocks)
 - Produce cleaner console output showing the KoreBuild lifecycle

Implementation challenges:
 - Acquisition. Loggers must be specified before NuGet restore
 - Nested builds. It gets messy
 - Interaction with console logger. In simply builds, it is easiest to let console logger do what it does by default, and supplement console output with info.  In parallel builds, we'd get cleaner console output if we override console logging and wrap all console output with a flow ID.

**Comment by natemcmaster** [comment]
Closing in favor of https://github.com/aspnet/BuildTools/issues/345
